### PR TITLE
feat: Collection orders with merkle tree for IDs

### PR DESCRIPTION
**Pull request by 0xShisui**

### CODE REVIEW

File: src/LooksRare.ts [hunk 1]
```diff
@@ -234,6 +236,48 @@ export class LooksRare {
     };
   }
 
+  /**
+   * Create a maker bid for collection offer.
+   * @see this.createMakerBid
+   * @param orderInputs Order data
+   * @returns CreateMakerBidOutput
+   */
+  public createMakerCollectionOffer(orderInputs: CreateMakerCollectionOfferInput): Promise<CreateMakerBidOutput> {
```
Comment (0xhiroshi) [inline src/LooksRare.ts:245]: is this an async function? why does it return a promise?

File: src/LooksRare.ts [hunk 2]
```diff
@@ -234,6 +236,48 @@ export class LooksRare {
...
+
+  public async createMakerCollectionOfferWithMerkleTree(
+    orderInputs: CreateMakerCollectionOfferInput,
+    itemIds: BigNumberish[]
+  ): Promise<CreateMakerBidOutput & { proofs: { itemId: BigNumberish; proof: string[] }[] }> {
+    const leaves = itemIds.map((itemId) => {
+      const hash = utils.keccak256(utils.solidityPack(["uint256"], [itemId]));
+      return Buffer.from(hash.slice(2), "hex");
+    });
+    const tree = new MerkleTreeJS(leaves, keccak256, { sortPairs: true });
```
Comment (0xhiroshi) [inline src/LooksRare.ts:257]: I think you might be able to `hashLeaves: true` instead of hashing the leaves by yourself

File: src/LooksRare.ts [hunk 3]
```diff
@@ -234,6 +236,48 @@ export class LooksRare {
...
+    const root = tree.getHexRoot();
+
+    const output = await this.createMakerBid({
+      ...orderInputs,
+      strategyId: StrategyType.collectionWithMerkleTree,
+      additionalParameters: [root],
+      itemIds: [],
+    });
+
+    const proofs = itemIds.map((itemId, index) => {
```
Comment (0xhiroshi) [inline src/LooksRare.ts:267]: proofs are only required during taker execution so I wonder if it is necessary at this point?

Comment (0xShisui) [reply src/LooksRare.ts:274]: If I don't compute it here, it mean we need to pass the full list of ids to the create taker function, not sure what's best

Comment (0xShisui) [reply src/LooksRare.ts]: Updated, I think it's better 👍

**Review by 0xhiroshi — APPROVED** [review]

File: src/LooksRare.ts (hunk 2 above)
Comment (0xShisui) [reply src/LooksRare.ts:271]: didn't work

**Pull request merged**